python/mxnet/ndarray/contrib.py

```
@@ -192,3 +192,126 @@ def check_input(inputs, in_type, msg):
+    until the loop condition is not satisfied.
+
+    ``cond'' is a user-defined loop condition function that takes ``loop_vars''
+    as input and return a boolean scalar ndarray to determine the termination of the loop.
+    The loop terminates when ``cond'' is false.
+
+    ``func'' is a user-defined loop body function that takes ``loop_vars'' as input,
+    performs computation and returns:
+    1) a list of NDArrays, which has the same number of NDArrays and the same types as ``loop_vars''
+    2) optionally outputs for each step
+    The signature of ``fun'' can be:
+       * def func(loop_vars): new_loop_vars
+    or * def func(loop_vars): (output, new_loop_vars)
+
+    ``loop_vars'' is a list of NDArrays that represent loop variables.
+
+    ``max_iterations'' is a python or NDArray scalar that defines the maximal number of iterations.
+    The maximal number of iterations is defined statically when the computation is constructed.
+
+    The computation done by this operator is equivalent to the pseudo code below
+    when the loop variables are NDArray:
+
+    # ``func'' returns only the new_loop_vars
+    steps = 0
+    while steps < max_iterations and cond(loop_vars):
+        loop_vars = func(loop_vars)
+        steps += 1
+    return loop_vars
+
+    # ``func'' returns (output, new_loop_vars)
+    steps = 0
+    outputs = []
+    while steps < max_iterations and cond(loop_vars):
+        output, loop_vars = func(loop_vars)
+        outputs.append(output)
+        steps += 1
+    return stack(outputs), loop_vars
+
+    TODO(Junru): review the documentation, it has been out-of-date.
+    Corner case #1: what if cond is always false? Do we return "outputs"?
+    """
+    def _to_python_type(inputs, type, name):
+        """Converts "inputs", possibly typed mxnet NDArray, a numpy ndarray, other python types,
+        to the given type
+        """
+        if isinstance(inputs, ndarray.NDArray):
+            inputs = inputs.asscalar()
```

do you need to cast the type here as well?

---

At users' convenience, if they pass NDArray into this function, we silently convert it to python scalar if possible. (This case happens very often in ML codes)

---

solved